src/openforms/submissions/tests/test_tasks_pdf.py

```diff
@@ -67,6 +67,96 @@ def test_pdf_renders_without_errors(self):
 
         capture.check()
 
+    def test_hidden_output_not_included(self):
+        """
+        Assert that hidden components are not included in the report.
+
+        #1451 requires that components hidden (statically or) dynamically by logic
+        are not included. The report functionality is also extended to add more
+        structure, and step titles/headers may not be included if there's nothing
+        visible in the step.
+        """
+        submission = SubmissionFactory.from_components(
+            [
+                {
+                    "key": "input1",
+                    "label": "Input 1",
+                    "type": "textfield",
+                    "hidden": True,
+                }
+            ],
+            # a multiple email field with no value entered sends a [None]
+            submitted_data={},
+            completed=True,
+            with_report=True,
+        )
+        html = submission.report.generate_submission_report_pdf()
+        step_title = submission.submissionstep_set.get().form_step.form_definition.name
+
+        # hidden component itself may not be visible
+        self.assertNotIn("Input 1", html)
+        # step has no visible children -> also not visible
+        self.assertNotIn(step_title, html)
+
+    def test_visible_output_included(self):
+        """
+        Assert that hidden components are not included in the report.
+
+        #1451 requires that components hidden (statically or) dynamically by logic
```

Comment:
Should there maybe also be a test that fields hidden by logic also are not shown?

Reply:
I don't think so - the low-level interaction of logic is tested in the Renderer/Node unit tests, and the effect is that they set the `hidden` property, which is simulated here with static definitions.